performjoin.go

```diff
@@ -22,9 +22,10 @@ type PerformJoinInput struct {
 	KeyID      KeyID              // Used to sign the join event
 	KeyRing    *KeyRing           // Used to verify the response from send_join
 
-	EventProvider   EventProvider        // Provides full events given a list of event IDs
-	UserIDQuerier   spec.UserIDForSender // Provides userID for a given senderID
-	SenderIDCreator spec.CreateSenderID  // Creates new senderID for this room
+	EventProvider             EventProvider                  // Provides full events given a list of event IDs
+	UserIDQuerier             spec.UserIDForSender           // Provides userID for a given senderID
+	SenderIDCreator           spec.CreateSenderID            // Creates new senderID for this room
```

Comment:
This could use a better name. It's not just a creator. It will create one if none exists, but otherwise it will return the existing senderID.

Reply:
And the comment should be updated as well